Spanish translation

venikunche opened the pull request.

Commit:
added Spanish as an option in the language settings

Commit:
added Spanish as an option in the language settings

Commit:
added Spanish as an option in the language settings

Commit:
added Spanish as an option in the language settings

Commit:
added translation for base.html

Commit:
added translation for base.html

Commit:
added translation for base.html

Commit:
added translation for base.html

Commit:
added translation for nav.html

Commit:
added translation for nav.html

Commit:
added translation for nav.html

Commit:
added translation for nav.html

Commit:
added translation for nav.html

Commit:
added translation for index.html

Commit:
added translation for index.html

Commit:
added translation for index.html

Commit:
added translation for index.html

Commit:
added translation for 404.html

Commit:
added translation for 404.html

Commit:
added translation for 404.html

Commit:
added translation for 404.html

Commit:
added translation for parents.html

Commit:
added translation for parents.html

Commit:
added translation for parents.html

Commit:
added translation for parents.html

Commit:
added translation for rules.html

Commit:
added translation for rules.html

Commit:
added translation for rules.html

Commit:
added translation for rules.html

Commit:
added translation for choose_adventure.html

Commit:
added translation for choose_adventure.html

Commit:
added translation for choose_adventure.html

Commit:
added translation for choose_adventure.html

Commit:
added translation for nature_walk_end.html

Commit:
added translation for nature_walk_end.html

Commit:
added translation for nature_walk_end.html

Commit:
added translation for nature_walk_end.html

Commit:
added translation for nature_walk_end.html

Commit:
added translation for nature_walk.html

Commit:
added translation for nature_walk.html

Commit:
added translation for nature_walk.html

Commit:
added translation for nature_walk.html

Commit:
added translation for start.html

Commit:
added translation for start.html

Commit:
added translation for start.html

Commit:
added translation for start.html

Commit:
added translation for time_travel.html

Commit:
added translation for time_travel.html

Commit:
added translation for time_travel.html

Commit:
added translation for time_travel.html

Commit:
added translation for time_travel_end.html

Commit:
added translation for time_travel_end.html

Commit:
added translation for time_travel_end.html

Commit:
added translation for time_travel_end.html

Commit:
added translation for swimming.html

Commit:
added translation for swimming.html

Commit:
added translation for swimming.html

Commit:
added translation for swimming.html

Commit:
added translation for swimming_end.html

Commit:
added translation for swimming_end.html

Commit:
added translation for swimming_end.html

Commit:
added translation for swimming_end.html

Commit:
added translation for first-game/start.html

Commit:
added translation for first-game/start.html

Commit:
added translation for first-game/start.html

Commit:
added translation for first-game/start.html

Commit:
added translation for infographic.html

Commit:
added translation for infographic.html

Commit:
added translation for infographic.html

Commit:
added translation for infographic.html

Commit:
added translation for endangered.html

Commit:
added translation for endangered.html

Commit:
added translation for endangered.html

Commit:
added translation for endangered.html

Commit:
added translation for lands_map.html

Commit:
added translation for lands_map.html

Commit:
added translation for lands_map.html

Commit:
added translation for lands_map.html

Commit:
added Spanish Activity Guides

Commit:
added Spanish Activity Guides

Commit:
added Spanish Activity Guides

Commit:
added Spanish Activity Guides

Commit:
added translation for how-it-works/index.html

Commit:
added translation for how-it-works/index.html

Commit:
added translation for how-it-works/index.html

Commit:
Merge remote-tracking branch 'origin/spanish-translation' into spanish-translation

Commit:
added translations for plan-your-trip

Commit:
added translations for plan-your-trip

Commit:
added translations for plan-your-trip

Commit:
added translations for get-your-pass

Commit:
added translations for get-your-pass

Commit:
added translations for get-your-pass

Commit:
added translation for legal/privacy.html

Commit:
added translation for legal/privacy.html

Commit:
added translation for legal/privacy.html

Commit:
added translation for about.html

Commit:
added translation for about.html

Commit:
added translation for about.html

Commit:
added translation for get-your-pass/game_success.html

Commit:
added translation for get-your-pass/game_success.html

Commit:
added translation for get-your-pass/game_success.html

Commit:
added translation for get-your-pass/fourth_grade_voucher.html

Commit:
added translation for get-your-pass/fourth_grade_voucher.html

Commit:
added translation for get-your-pass/fourth_grade_voucher.html

Commit:
added translation for field trip sites page for when there are no results

Commit:
added translation for field trip sites page for when there are no results

Commit:
added translation for field trip sites page for when there are no results

Commit:
removed extra space in "General Services Administration"

Commit:
removed extra space in "General Services Administration"

Commit:
removed extra space in "General Services Administration"

Commit:
added translation for image alt tags

Commit:
added translation for image alt tags

Commit:
added translation for image alt tags

Commit:
added a link to view the site in English or Spanish

Commit:
added a link to view the site in English or Spanish

Commit:
added a link to view the site in English or Spanish

Commit:
Updated Django due to security patch.

Commit:
Merge pull request #256 from 18F/security-version-update

Updated Django due to security patch.

Commit:
Added cap for Statistics query when analyzing educator requests

Commit:
More number/total corrections

Commit:
student pass calculation correction

Commit:
Merge pull request #258 from 18F/educator_number_cap

Educator number cap

OpenGlobe committed:
Update stats.html

Removes student from descriptor.

OpenGlobe committed:
Update stats.html

- removed student from descriptor
- added some detail explaining difference in numbers

Commit:
Merge pull request #259 from 18F/stats-ed-numbers-3

Updates to stats page

mogul committed:
Cloud Foundry forward-compatibility

Per [Cloud Foundry documentation](https://docs.cloudfoundry.org/running/apps-enable-diego.html#app-code) use of `$VCAP_APP_HOST` and `$VCAP_APP_PASSWORD` are incompatible with newer (Diego-based) Cloud Foundry deployments. This change makes this repository forward compatible with the new cloud.gov environment deployment in AWS GovCloud.

npstcole committed:
Merge remote-tracking branch '18F/master'

npstcole committed:
Uncomment Amtrak Announcement

khandelwal committed:
Merge pull request #261 from trevorcolenps/master

Master

Commit:
Merge pull request #260 from mogul/patch-1

Cloud Foundry forward-compatibility

Commit:
Merge remote-tracking branch 'origin/spanish-translation' into spanish-translation

Commit:
translation link fix

fixed the translation link so that it doesn’t show up as a button

Commit:
translation link fix

fixed the translation link so that it doesn’t show up as a button

Commit:
translation updates

fixed translation and added missing translations

Commit:
translation updates

fixed translation and added missing translations

Commit:
translation link in the responsive version

added translation link in the menu and home menu sections of the
responsive version of the site

Commit:
translation link in the responsive version

added translation link in the menu and home menu sections of the
responsive version of the site

Commit:
Fix 'Go climbing' translation

Commit:
Fix 'Go climbing' translation

Commit:
translation adjustments

Commit:
translation adjustments

Commit:
capitalized the locale directories

Commit:
capitalized the locale directories

Commit:
translation corrections

Commit:
translation corrections

Commit:
compiled language files

Commit:
compiled language files

Commit:
removed compiled translation files from being ignored

Commit:
removed compiled translation files from being ignored

Commit:
Fix api issues

Commit:
Remove print statement

Commit:
Merge pull request #262 from khandelwal/fix_api

Fix failing tests around the record locator API

khandelwal commented:
Hello! If you update this pull request (and fix the conflicts as well) - I think the tests should now pass. We pushed out a pull request that fixes the breaking tests.

Commit:
Merge branch 'spanish-translation' of https://github.com/USGS-WiM/ekip-api into spanish-translation

Commit:
removed caching

Commit:
fixed the translation link in mobile site

Commit:
recompiled the language files

Commit:
recompiled the language files

Commit:
fixed educator pass preview form

Commit:
fixed the translation link for the educator pass page

khandelwal commented on ekip/config/settings/base.py:
```diff
@@ -90,6 +92,11 @@
 
+LANGUAGES = (
+    ('en-us', _('English')),
+    ('es-us', _('Spanish')),
```
Should the label here actually say 'Spanish' in Spanish?

venikunche replied:
No. I've put all of the Spanish translations in the .po files.

khandelwal commented on ekip/everykid/static/css/voucher.css:
```diff
@@ -31,7 +31,7 @@ html, body {
 .voucher-container, .voucher-instructions {
   width: 610px;
-  height: 842px;
+  height: 880px;
```
Why did this change based on a language change?

venikunche replied:
The Spanish text takes more space. With the 842px height, the text was running out of the border like this:
<img width="656" alt="screen shot 2016-11-30 at 10 44 37 am" src="https://cloud.githubusercontent.com/assets/840844/20759127/092395f0-b6ea-11e6-856f-10cc977ca6c1.png">

khandelwal commented on ekip/everykid/templates/get-your-pass/educator_passes.html:
```diff
@@ -41,30 +41,29 @@
-      <p>Activity 3: <a href="{% static "activities/scholastic-three.pdf" %}" target="_blank">Our Nation’s Native Peoples</a> (PDF). This lesson teaches students about the people who lived on this land before it was called the United States.</p>
-      <p>Activity 4: <a href="{% static "activities/scholastic-four.pdf" %}" target="_blank">Citizen Science</a> (PDF). This lesson helps kids learn about the difference between weather and climate.</p>
+      <h3>{% trans "Download" %}<br/> {% trans "Activity Guides" %}</h3>
+      <p>{% trans "Activity" %} 1: <a href="{% if LANGUAGE_CODE == 'es-us' %}{% static "activities/es_us/scholastic-one.pdf" %}{% else %}{% static "activities/scholastic-one.pdf" %}{% endif %}" target="_blank">{% trans "Exploring Federal Lands and Waters" %}</a> (PDF). {% trans "This lesson teaches students why our country protects lands and waters." %}</p>
```
In the 'URL' for the PDF, I might use the LANGUAGE_CODE parameter, instead of hard-coding 'es-us'

venikunche replied:
That makes sense. I'll make the change.

khandelwal commented on ekip/everykid/templates/about.html:
```diff
@@ -141,23 +123,23 @@
-      <p>Here's a <a href="{% static "media/tool-kit-2.zip" %}">set of resources</a> (ZIP file, 15.4 MB) to help local officials, youth leaders and others plan an outdoor outing with 4th graders.</p>
-      <p>In partnership with the Every Kid in a Park Program, Transforming Youth Outdoors <a href="https://tyo.pathgather.com/l/login" target="_blank">has created two step-by-step learning paths</a> that give classroom teachers and outdoor educators tools to learn more about the Every Kid program and plan a successful field trip. These resources provide information about federal lands and waters, ...
+      <p>{% trans "Here's a" %} <a href="{% static "media/tool-kit-2.zip" %}">{% trans "set of resources" %}</a> (ZIP file, 15.4 MB) {% trans "to help local officials, youth leaders and others plan an outdoor outing with 4th graders." %}</p>
+      <p>{% trans "In partnership with the Every Kid in a Park Program, Transforming Youth Outdoors" %} <a href="https://tyo.pathgather.com/l/login" target="_blank">{% trans "has created two step-by-step learning paths" %}</a> {% trans "that give classroom teachers and outdoor educators tools to learn more about the Every Kid program and plan a successful field trip. These resources provide infor ...
```
I would recommend wrapping long text lines such as these, so that it's a bit easier to read.

venikunche replied:
Unfortunately, when using {% trans "text" %}, all of the text had to be in one line. The translations I was given were by paragraphs. Since the structure of a sentence in English and Spanish may be different. I wasn't sure on how to break them up.

khandelwal commented on ekip/everykid/templates/index.html:
```diff
@@ -29,23 +30,35 @@
-          <a href="{% url 'parents' %}">Parents</a>
-        </li>
+          <a href="{% url 'parents' %}">{% trans "Parents" %}</a>
+        </li>	
```
This form seems to be duplicated in multiple locations. With template inheritance, I think you shouldn't need to to this. At the least, you should be able to type it out once, and then just do an include.

venikunche replied:
Agreed. For now, I focused on just translating the site and tried not to change the original site unless necessary.

venikunche replied:
I misunderstood your comment earlier. Yes, it makes sense to type it out once and then just do an include.

khandelwal commented on ekip/everykid/templates/nav.html:
```diff
@@ -1,15 +1,28 @@
       </li>
       <li class="{% active request 'about' %}">
-        <a href="{% url 'about_ekip' %}">About</a>
+        <a href="{% url 'about_ekip' %}">{% trans "About" %}</a>
```
Repeated form.

khandelwal commented on requirements.txt:
```diff
@@ -1,13 +1,13 @@
 # Django, using PostgreSQL
-Django==1.9.6
+Django==1.9.10
```
This should not have changed from Master. Is this pull request updated?

venikunche replied:
The pull request is updated. It looks like it is the right version. The [master](https://github.com/18F/ekip-api/blob/c1b20b40a3a10e9876e68be7537ed2709ff945b9/requirements.txt) also has 1.9.10.

khandelwal commented on ekip/everykid/urls.py:
```diff
@@ -30,36 +28,39 @@
         name="educator_passes"),
-    url(r'get-your-pass/', cache_page(60*60)(TemplateView.as_view(
-        template_name="get-your-pass/index.html")), name="get_your_pass"),
+    url(r'get-your-pass/', TemplateView.as_view(
```
Removing all this caching is making me nervous. Before this hits production, I think you need to look at how to make the CSRF stuff work with caching. I believe there's some documentation on the Django website that talks to this.

venikunche replied:
Sounds good. I'll take a look.

khandelwal commented on ekip/redemption/views.py:
```diff
@@ -202,15 +202,17 @@ def tables(request):
 
-    if not educator_tickets:
-        educator_tickets = 0
+    if not educator_tickets_cleaned:
```
How is this change related to the Spanish translation?

venikunche replied:
Not sure on what's going on here. I did not make any changes to this file. I looked at [our repository](https://github.com/USGS-WiM/ekip-api/commits/spanish-translation/ekip/redemption/views.py) and it hasn't been modified. I compared it with what's in master and it's the same. I am not sure on why GitHub is showing it as though there is a difference.

Commit:
added back caching

added the ensure_csrf_cookie decorator which forces setting of the
translation form’s cookie.

Commit:
setup the repeat translation forms as an include

Commit:
used LANGUAGE_CODE parameter instead of hard-coding it
